src/main/node/file-manager.ts

```diff
@@ -0,0 +1,86 @@
+import fs from 'fs';
+import path from 'path';
+import { addPublication } from '../../shared/slices/publicationsSlice';
+
+let configFileName = 'vn-pub.conf';
+
+function isRepository(source: string): boolean {
+  return fs.existsSync(path.join(source, '.git'));
+}
+
+/**
+ * Checks if given path is a publication. Returns true if there is a config file in the directory.
+ * @param {string} source
+ * @return {boolean}
+ */
+export function isPublication(source: string): boolean {
+  return fs.existsSync(path.join(source, configFileName));
+}
+
+/**
+ * Checks if given path is a directory
+ * @param {string} source
+ * @return {boolean}
+ */
+export function isDirectory(source: string): boolean {
+  try {
+    return fs.lstatSync(source).isDirectory();
+  } catch (error) {
+    return false;
+  }
+}
+
+function getDirectories(source: string): string[] {
+  let directories: string[] = [];
+  try {
+    const isDirectory = (source: string) => fs.lstatSync(source).isDirectory();
+    directories = fs
+      .readdirSync(source)
+      .map((name) => path.join(source, name))
+      .filter(isDirectory);
+  } catch (error) {}
+  return directories;
+}
+
+function recursiveSearch(source: string): void {
+  if (isPublication(source)) {
+    try {
+      let filePath = path.join(source, configFileName);
+      let rawdata = fs.readFileSync(filePath);
+      let dataParsed = JSON.parse(rawdata.toString());
+      addPublication({
+        project_name: dataParsed.project_name,
+        collaborators: dataParsed.collaborators,
+        pm_preference: dataParsed.collaborators,
+        description: dataParsed.description,
+        dirPath: source,
+      });
+    } catch (err) {}
+  } else if (!isRepository(source)) {
+    const availableDirectories: string[] = getDirectories(source);
+    for (let i = 0; i < availableDirectories.length; i++) {
+      recursiveSearch(availableDirectories[i]);
+    }
+  }
+}
+
+/**
+ * Finds all publication config files and uploads their content to the application state.
+ * Rejects the promise if a command execution error occurs.
+ * @param {string} source
+ * @return {Promise<boolean>}
+ */
+export function findLocalPublications(source: string): Promise<boolean> {
```

Comment:
This is not asynchronous in any way right? Does it affect the application / stop it for a significant amount of time?

Reply:
I thought I can just use `return new Promise()` to make it asynchronous. Do I have to make it "async, await"?

Reply:
@lukasz-starosta It searches for publications recursively in a folder structure. All file access operations must be async, since it can take a lot of time, either because of the number of folders to search or having slow data access (network storage over SFTP, HDDs etc.).